set/set.go

```diff
@@ -18,24 +18,26 @@ package set
 
 import (
 	"sort"
+
+	"k8s.io/utils/genericinterfaces"
 )
 
 // Empty is public since it is used by some internal API objects for conversions between external
 // string arrays and internal sets, and conversion logic requires public types today.
 type Empty struct{}
 
 // Set is a set of the same type elements, implemented via map[ordered]struct{} for minimal memory consumption.
-type Set[E ordered] map[E]Empty
+type Set[E genericinterfaces.Ordered] map[E]Empty
 
 // New creates a new set.
-func New[E ordered](items ...E) Set[E] {
+func New[E genericinterfaces.Ordered](items ...E) Set[E] {
 	ss := Set[E]{}
 	ss.Insert(items...)
 	return ss
 }
 
 // KeySet creates a Set[E] from a keys of a map[E](? extends interface{}).
-func KeySet[E ordered, A any](theMap map[E]A) Set[E] {
+func KeySet[E genericinterfaces.Ordered, A any](theMap map[E]A) Set[E] {
```

Comment:
What's the motivation for exporting the type? are we expecting other things to reference the interface? They couldn't previously, which had the benefit of only allowing people to declare KeySet with their own types, which had to be compatible with our constraints... that seemed reasonable to me.

Reply:
I'd need to redeclare the exact same type in the maps package otherwise, since the output of the value of a map is a generic set.

Reply:
How about waiting for the bump to Go 1.21 in utils and relying on `cmp.Ordered`? See also #295.